### PR TITLE
Task/#104038 endpoint get site with application and allocations

### PR DESCRIPTION
**Pull request by PawelBargielski**
### 🛠 Changes being made

Summarize the changes you've made in this pull request.

### 📸 Screenshots (optional)

Insert some screenshots if you made UI changes.

### 🏎 Quality check

- [ ] Have you performed local tests?
- [ ] Are integration tests passing locally?
- [ ] Have you added some unit tests?
- [ ] Have you added some integration tests?
- [ ] Have you checked WCAG (included screenshot)

### CODE REVIEW

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            return ExternalUserId != null && OrganizationId != null && SiteId != null;
+        }
+
+        public override void DoWork()
+        {
+            TracingService.Trace("Get consortiumId");
+            string consortiumId = null;
+            if (_consortiumService.GetConsortiumIdForSite(SiteId) != Guid.Empty)
+            {
+                consortiumId = _consortiumService.GetConsortiumIdForSite(SiteId).ToString();
```
Comment (jjankowski87) [inline src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs:55]: extract variable, right now there are 2 requests for fetching Consortium

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                if (con != Guid.Empty)
+                {
+                    TracingService.Trace($"Set value - consortium");
+                    TracingService.Trace($"ConsorciumId : {con}");
+                    consortiumId = con.ToString();
+                }
+
+                var application = new List<invln_scheme> { app };
+                var applicationsDict = application.ToDictionary(k => k.invln_contactid);
+                var webroleList = _contactWebroleRepository.GetListOfUsersWithoutLimitedRole(OrganizationId.ToString());
```
Comment (jjankowski87) [inline src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs:104]: webroles and contacts can be fetched outside loop

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+                var con = _consortiumService.GetConsortiumIdForApplication(app.Id);
+                if (con != Guid.Empty)
+                {
+                    TracingService.Trace($"Set value - consortium");
+                    TracingService.Trace($"ConsorciumId : {con}");
+                    consortiumId = con.ToString();
+                }
+
+                var application = new List<invln_scheme> { app };
```
Comment (Carq) [inline src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs:102]: Do you need to create list of one element and later (line 107) iterate on the list?

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationHandler.cs [hunk 4]
```diff
@@ -72,7 +72,7 @@ public override void DoWork()
             TracingService.Trace("Filter Application");
             var applications = _ahpApplicationRepository.GetByAttribute(invln_scheme.Fields.invln_Site, site.Id)
                                 .Where(x => x.invln_ExternalStatus.Value != (int)invln_ExternalStatusAHP.Deleted &&
-                                x.invln_contactid != null).ToList()
+                                x.invln_contactid != null && x.invln_isallocation == false && x.invln_IsLatestVersion == true).ToList()
```
Comment (Carq) [inline src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationHandler.cs:75]: What happen if site has only allocation? Will the site be returned?

Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationHandler.cs:75]: Yes, site be returned.

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs (hunk 1 above)
Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs]: I forgot that I get consortium here.
I had it fixed in the loop, but I forgot here.
Done.

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs (hunk 2 above)
Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs]: You're right.

WebRole yes.
Contact no. Contact is get per application.

Done

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs (hunk 3 above)
Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs]: Of course you are right.
It is a legacy code.
I did not understand why it was done this way here.
And I left it as it was.

Fixed!

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            if (consortiumId == Guid.Empty.ToString())
+            {
+                consortiumId = null;
+            }
+
+            TracingService.Trace("GetSiteApplicationHandler Do Work");
+            if (!_consortiumService.CheckAccess(Operation.Get, RecordType.Site, ExternalUserId, SiteId.ToString(), null, consortiumId, OrganizationId.ToString()))
+                return;
+            TracingService.Trace("Get Site");
+            var site = _siteRepository.GetById(new Guid(SiteId.ToString()));
```
[the remark this answers is not in the document]
Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs]: Change made

File: src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        }
+
+        private List<invln_scheme> RecordsFiltering(List<invln_scheme> applications, invln_Sites site, string consortiumId, Dictionary<EntityReference, invln_contactwebrole> webroleDict)
+        {
+            var filteredApplication = new List<invln_scheme>();
+            TracingService.Trace($"Iteration by application no: {applications.Count}");
+            foreach (var app in applications)
+            {
+                TracingService.Trace($"Excluding records from the list, which are for a Limited User.");
+                var contact = _contactRepository.GetById(app.invln_contactid.Id, new string[] { Contact.Fields.FirstName, Contact.Fields.LastName, nameof(Contact.invln_externalid).ToLower() });
```
[the remark this answers is not in the document]
Comment (PawelBargielski) [reply src/HE.CRM.AHP.Plugins/Handlers/CustomApi/FrontDoor/GetSiteApplicationAllocationHandler.cs]: Change made

**Review by Carq — APPROVED** [review]

**Pull request merged**